SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -764,31 +764,30 @@ void CEulerSolver::Set_MPI_ActDisk(CSolver **solver_container, CGeometry *geomet
 
 void CEulerSolver::Set_MPI_Nearfield(CGeometry *geometry, CConfig *config) {
 
-  unsigned long iter,  iPoint, iVertex, jVertex, iPointTotal,
+  unsigned long iter,  iPoint, iVertex, jVertex, iPointTotal, irecv,
   Buffer_Send_nPointTotal = 0;
   long iGlobalIndex, iGlobal;
   unsigned short iVar, iMarker, jMarker;
   long nDomain = 0, iDomain, jDomain;
 
-  /*--- MPI status and request arrays for non-blocking communications ---*/
+  /*--- MPI request arrays for non-blocking communications ---*/
 
-  SU2_MPI::Status status;
-  SU2_MPI::Request req;
+  SU2_MPI::Request *req_Recv   = new SU2_MPI::Request[size-1];
 
   /*--- Define buffer vector interior domain ---*/
 
-  su2double        *Buffer_Send_PrimVar          = nullptr;
-
-  unsigned long *nPointTotal_s = new unsigned long[size];
-  unsigned long *nPointTotal_r = new unsigned long[size];
-  su2double     *iPrimVar      = new su2double [nPrimVar];
-
-  unsigned long Buffer_Size_PrimVar = 0;
-  unsigned long PointTotal_Counter = 0;
+  su2double     *Buffer_Send_PrimVar = nullptr;
+  unsigned long *nPointTotal_s       = new unsigned long[size];
+  unsigned long *nPointTotal_r       = new unsigned long[size];
+  su2double     *iPrimVar            = new su2double [nPrimVar];
+  unsigned long nPointTotal_rCum[size+1];
```

Comment:
You have to make this one a vector<unsigned long>, as variable-length arrays are not standard c++.

Reply:
I had trouble using vectors, so I used arrays. I'll try to work on it tomorrow.

Reply:
Actually, I was able to fix them easily.